File: app/src/main/java/com/nexaas/pkceclienttest/interfaces/OauthTokenService.java
package com.nexaas.pkceclienttest.interfaces;

import com.nexaas.pkceclienttest.models.NexaasIdUser;
import com.nexaas.pkceclienttest.models.OauthToken;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

/**
 * Created by alexandre on 23/08/2018.
 * Copyright© Nexaas™. All rights reserved.
 */

public interface OauthTokenService {

    @POST ("/oauth/token")
    Call<NexaasIdUser> oauthToken (@Body OauthToken oauthToken);
}

File: nexaasidauth/src/main/java/com/nexaas/nexaasid/auth/api/services/OauthTokenService.kt
package com.nexaas.nexaasid.auth.api.services

import com.nexaas.nexaasid.auth.model.OAuthTokenRequest
import com.nexaas.nexaasid.auth.model.OAuthTokenResponse
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.GET
import retrofit2.http.Query

interface OauthTokenService {

    @POST("oauth/token")
    fun getOAuthTokenResponse(@Body oauthTokenRequest: OAuthTokenRequest): Call<OAuthTokenResponse>

    @GET("applications/sign_out")
    fun signOut(@Query("access_token") accessToken: String): Call<Void>
}